### PR TITLE
Make some macros indexable #294

### LINKED ISSUE #294: Make `ChannelId` indexable
I can't even find it manually, would it be possible to just define the base struct without whatever (I'm assuming) macro?

### PR DESCRIPTION
**Pull request by ForsakenHarmony**
You can reject it if you don't like it

This also includes some shorthands for struct creation

Closes #294

### CODE REVIEW

File: src/voice/audio.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ pub const SAMPLE_RATE: u32 = 48_000;
 
 /// A readable audio source.
 pub trait AudioSource: Send {
-    fn is_stereo(&mut self) -> bool;
```
Comment (acdenisSK) [inline src/voice/audio.rs]: Irrelevant [to this pr] and (possibly?) an unnecessary breaking change. So revert this please.

File: src/model/channel/message.rs [hunk 2]
```diff
@@ -14,6 +14,7 @@ use std::fmt::Write;
 use std::mem;
 #[cfg(feature = "model")]
 use {constants, http, utils as serenity_utils};
+use std::{i64, u64};
```
Comment (acdenisSK) [inline src/model/channel/message.rs]: Unnecessary imports.

File: src/internal/timer.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 use chrono::{DateTime, Duration, Utc};
 use std::time::Duration as StdDuration;
 use std::thread;
+use std::{i64};
```
Comment (acdenisSK) [inline src/internal/timer.rs]: Unnecessary import.

File: src/client/bridge/gateway/shard_messenger.rs [hunk 4]
```diff
@@ -3,6 +3,7 @@ use model::prelude::*;
 use super::{ShardClientMessage, ShardRunnerMessage};
 use std::sync::mpsc::{SendError, Sender};
 use websocket::message::OwnedMessage;
+use std::result::Result;
```
Comment (acdenisSK) [inline src/client/bridge/gateway/shard_messenger.rs]: Unnecessary import.

File: src/builder/create_embed.rs [hunk 5]
```diff
@@ -26,6 +26,7 @@ use utils::VecMap;
 
 #[cfg(feature = "utils")]
 use utils::Colour;
+use std::u64;
```
Comment (acdenisSK) [inline src/builder/create_embed.rs]: Unnecessary import.

File: src/internal/timer.rs (hunk 3 above)
Comment (ForsakenHarmony) [reply src/internal/timer.rs]: Idk why but my ide was complaining when it wasn't imported, but I can remove it again

File: src/voice/audio.rs (hunk 1 above)
Comment (ForsakenHarmony) [reply src/voice/audio.rs]: I mean it should just read a value, and the only use does that, but you're right it's unrelated

Comment (ForsakenHarmony) [reply src/voice/audio.rs]: that's why I did it

**Comment by ForsakenHarmony** [comment]
Something seems to be wrong with travis?

**Pull request merged**